### PR TITLE
fix(analytics-js): use globalThis instead of window to fix issues in SSR

### PR DESCRIPTION
**Pull request by bardisg**
## PR Description

Refactor to ensure the generated npm package will work when imported as dependency to fraeworks that support SSR like NextJS.

- fixes on types to allow reuse and reduce the shared bundles that are generated by modeule federation
- use globalThis instead of window
- ensure no calls to initialize window APIs (like cookie storage) happens before the load method is called
- fixes on html templates
- bumped most dependencies to the latest

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by saikumarrs — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

File: packages/analytics-js-plugins/src/utilities/retryQueue/index.ts [hunk 1]
```diff
@@ -327,14 +328,12 @@ class Queue extends Emitter {
...
-        id,
-        validKeys: QueueStatuses,
-      },
-      getStorageEngine('localStorage'),
-    );
+    const other = this.storeManager.setStore({
+      id,
+      name: this.name,
+      validKeys: QueueStatuses,
+      type: 'localStorage',
```
Comment (saikumarrs) [inline packages/analytics-js-plugins/src/utilities/retryQueue/index.ts:335]: This should also respect the `storageType` input in the constructor.

File: packages/analytics-js-plugins/src/utilities/retryQueue/index.ts [hunk 2]
```diff
@@ -462,14 +461,12 @@ class Queue extends Emitter {
...
-              id: parts[1],
-              validKeys: QueueStatuses,
-            },
-            getStorageEngine('localStorage'),
-          ),
+          this.storeManager.setStore({
+            id: parts[1],
+            name,
+            validKeys: QueueStatuses,
+            type: 'localStorage',
```
Comment (saikumarrs) [inline packages/analytics-js-plugins/src/utilities/retryQueue/index.ts:468]: Same as above.

File: packages/analytics-js-plugins/src/utilities/retryQueue/index.ts (hunk 1 above)
Comment (bardisg) [reply packages/analytics-js-plugins/src/utilities/retryQueue/index.ts:335]: on it

File: packages/analytics-js-plugins/src/utilities/retryQueue/index.ts (hunk 2 above)
Comment (bardisg) [reply packages/analytics-js-plugins/src/utilities/retryQueue/index.ts:468]: on it

File: packages/analytics-js-plugins/src/utilities/retryQueue/index.ts (hunk 1 above)
Comment (bardisg) [reply packages/analytics-js-plugins/src/utilities/retryQueue/index.ts:335]: seems that this hardcodded value is as expected asreclaim only happens from localstrage

File: packages/analytics-js-plugins/src/utilities/retryQueue/index.ts (hunk 2 above)
Comment (bardisg) [reply packages/analytics-js-plugins/src/utilities/retryQueue/index.ts:468]: seems that this hardcodded value is as expected asreclaim only happens from localstrage

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**